### PR TITLE
[cached] unset_interface_state implementation

### PR DESCRIPTION
**Pull request by emmurd**

### CODE REVIEW

File: netman/adapters/switches/cached.py [hunk 1]
```diff
@@ -291,6 +291,11 @@ def set_interface_state(self, interface_id, state):
         self.real_switch.set_interface_state(interface_id, state)
         self.interfaces_cache[interface_id].shutdown = (state == OFF)
 
+    def unset_interface_state(self, interface_id):
+        self.real_switch.unset_interface_state(interface_id)
+        interface = self.real_switch.get_interface(interface_id)
```
Comment (lindycoder) [inline netman/adapters/switches/cached.py]: You don't want to get the interface on real_switch to refresh your cache,
imagine unset is the only thing i do, why refresh the cache?

This call should invalidate the cache for this interface which will be refreshed upon request, not automatically
upon request = get_interface / get_interfaces
just add it to refresh items no?

File: tests/adapters/switches/cached_switch_test.py [hunk 2]
```diff
@@ -321,6 +321,22 @@ def test_set_interface_state_on(self):
...
 
+    def test_unset_interface_state(self):
+        self.real_switch_mock.should_receive("get_interfaces").once() \
+            .and_return([Interface('xe-1/0/2')])
+        self.switch.get_interfaces()
+
+        self.real_switch_mock.should_receive("unset_interface_state").once() \
+            .with_args('xe-1/0/2')
+
+        self.real_switch_mock.should_receive("get_interface").once() \
```
Comment (lindycoder) [inline tests/adapters/switches/cached_switch_test.py]: Before this line there should be a self.real_switch.get_interface... that will NOT use the cache and refetch it

**Comment by idjaw** [comment]
LGTM

File: tests/adapters/switches/cached_switch_test.py [hunk 3]
```diff
@@ -321,6 +320,23 @@ def test_set_interface_state_on(self):
...
+
+        self.real_switch_mock.should_receive("unset_interface_state").once() \
+            .with_args('xe-1/0/2')
+
+        self.switch.unset_interface_state('xe-1/0/2')
+
+        self.real_switch_mock.should_receive("get_interface").once() \
+            .with_args('xe-1/0/2') \
+            .and_return(Interface('xe-1/0/2', shutdown=False))
+
```
Comment (lindycoder) [inline tests/adapters/switches/cached_switch_test.py]: 2 lines here and you don't need all the line wrapping here, 120 columns ftw

**Comment by lindycoder** [comment]
:+1:

**Pull request merged**